pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1309,3 +1340,111 @@ func initializeAnnotations(cd *hivev1.ClusterDeployment) {
+		if err != nil {
+			errMsg := "unable to merge global pull secret with local pull secret"
+			cdLog.WithError(err).Error(errMsg)
+			return "", fmt.Errorf("%s : %v", errMsg, err)
+		}
+		return pullSecret, nil
+	case globalPullSecret != "":
+		return globalPullSecret, nil
+	case localPullSecret != "":
+		return localPullSecret, nil
+	default:
+		errMsg := "clusterdeployment must specify pull secret since hiveconfig does not specify a global pull secret"
+		cdLog.Error(errMsg)
+		return "", fmt.Errorf("%s", errMsg)
+	}
+}
+
+// updatePullSecretInfo adds pull secret information in cluster deployment and cluster deployment status
+func (r *ReconcileClusterDeployment) updatePullSecretInfo(pullSecret string, cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	var secretName string
+
+	// Get existing pull secret
+	isPullSecretObjExists := true
+	existingPullSecretObj := &corev1.Secret{}
+	if cd.Status.PullSecret.Name != "" {
+		secretName = cd.Status.PullSecret.Name
+		err := r.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: cd.Namespace}, existingPullSecretObj)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				isPullSecretObjExists = false
+			} else {
+				return fmt.Errorf(fmt.Sprintf("Error getting pull secret from cluster deployment : %v", err))
+			}
+		}
+	} else {
+		isPullSecretObjExists = false
+	}
+
+	if isPullSecretObjExists {
+		existingPullSecret, ok := existingPullSecretObj.Data[corev1.DockerConfigJsonKey]
+		if !ok {
+			return fmt.Errorf("Pull secret %s did not contain key %s", secretName, corev1.DockerConfigJsonKey)
+		}
+		hashOfexistingSecret, err := controllerutils.GetHash(existingPullSecret)
+		if err != nil {
+			return fmt.Errorf(fmt.Sprintf("Error creating checksum of %s : %v", secretName, err))
+		}
+		hashOfNewSecret, err := controllerutils.GetHash([]byte(pullSecret))
+		if err != nil {
+			return fmt.Errorf(fmt.Sprintf("Error creating checksum of new pull secret : %v", err))
+		}
+
+		// Check if the pull secrets checksums  are  different
+		if hashOfexistingSecret == hashOfNewSecret {
+			cdLog.Debug("No new pull secret found")
+			return nil
+		}
+	}
+
+	// delete the existing pull secret object
+	if isPullSecretObjExists {
+		err := r.Delete(context.TODO(), existingPullSecretObj, client.PropagationPolicy(metav1.DeletePropagationBackground))
+		if err != nil {
+			return fmt.Errorf(fmt.Sprintf("Error deleting the old existing pull secret %s: %v", secretName, err))
+		}
+	}
+
+	// create a new pull secret object and update the cluster deployment status
+	newSecretName := apihelpers.GetResourceName(cd.Name, "merged-pullsecret-"+utilrand.String(8))
+	newPullSecretObj := generatePullSecretObj(pullSecret, newSecretName, cd)
+	err := r.Create(context.TODO(), newPullSecretObj)
+	if err != nil {
+		return fmt.Errorf(fmt.Sprintf("error creating new pull secret object: %v", err))
+	}
+	cdLog.Debug("Created the new pull secret object in cluster deployment successfully")
```

Comment:
this should be Info level, and it should include a WithField("secretName", newPullSecretObj.Name)

Reply:
"created new merged pull secret for cluster deployment"